app/controllers/concerns/shopify_app/ensure_installed.rb

```diff
@@ -17,6 +17,7 @@ module EnsureInstalled
 
       before_action :check_shop_domain
       before_action :check_shop_known
+      before_action :validate_emebedded_session_is_active, if: :embedded_param?
```

Comment:
@paulomarg - I know you mentioned you wanted to use the `params[:embedded] == 1` flag here. I'm assuming here we want to use [all these conditionals from embedded_param as well?](https://github.com/Shopify/shopify_app/blob/main/lib/shopify_app/controller_concerns/redirect_for_embedded.rb#L14)

Reply:
Yeah, the idea is that we only need to run this check IF we're planning on redirecting to the admin (`embedded !== 1`).

This keeps the number of requests we need to make to a minimum because normally the app would just be loaded straight from the admin (`embedded == 1`), and for that to happen we know it's installed (Admin still references it).